tools/c7n_azure/c7n_azure/session.py

```diff
@@ -303,6 +314,17 @@ def get_functions_auth_string(self, target_subscription_id):
 
         return json.dumps(function_auth_params, indent=2)
 
+    def resolve_auth_endpoint(self, endpoint):
+        if endpoint == constants.VAULT_AUTH_ENDPOINT:
+            self.resource_namespace = get_keyvault_auth_endpoint(self.endpoints)
+            self.keyvault_auth_override = True
+
+        elif endpoint == constants.STORAGE_AUTH_ENDPOINT:
+            # These endpoints are not Cloud specific, but the suffixes are
+            self.resource_namespace = constants.STORAGE_AUTH_ENDPOINT
+        else:
+            self.resource_namespace = getattr(self.endpoints.endpoints, endpoint)
```

Comment:
the `self.endpoints.endpoints` stuff isn't ideal - per recommendation above I'd rename to something else, even `cloud_endpoints` so it makes a little more sense - I know it comes at expense of restrictive line length linting making life hard...